Client id mismatch after Cross-Origin-Opener-Policy Header

youennf opened the pull request:
#### dfeb33f0ac4c4f96cf0e09aca2ae8ffe0fd0ca1a
<pre>
Client id mismatch after Cross-Origin-Opener-Policy Header
<a href="https://rdar.apple.com/problem/149370100">rdar://problem/149370100</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=291315">https://bugs.webkit.org/show_bug.cgi?id=291315</a>

Reviewed by Brady Eidson.

Before the patch, in case of process swapping, we would create a new DocumentLoader in the final process.
This would mean a new ScriptExecutionContextIdentifier and so a new clientId that is not is the same as the inital DocumentLoader
that triggered the request sent to the service worker.

To fix this, when process swap happens in NetworkResourceLoader::transferToNewWebProcess, we tell the new DocumentLoader that it
should reuse the initial DocumentLoader ScriptExecutionContextIdentifier (with the right process identifier).

We update NetworkResourceLoader, WebResourceLoader and DocumentLoader accordingly.
We then have to deal in SWServer with ScriptExecutionContextIdentifier that have the same UUID but not the same process identifier.

* LayoutTests/http/tests/workers/service/resultingClientId-worker.js: Added.
* LayoutTests/http/tests/workers/service/resultingClientId.https-expected.txt: Added.
* LayoutTests/http/tests/workers/service/resultingClientId.https.html: Added.
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::setNewResultingClientId):
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/workers/service/server/SWServer.cpp:
(WebCore::SWServer::registerServiceWorkerClient):
(WebCore::SWServer::unregisterServiceWorkerClient):
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::transferToNewWebProcess):
* Source/WebKit/WebProcess/Network/WebResourceLoader.cpp:
(WebKit::WebResourceLoader::updateResultingClientIdentifier):
* Source/WebKit/WebProcess/Network/WebResourceLoader.h:
* Source/WebKit/WebProcess/Network/WebResourceLoader.messages.in:

Canonical link: <a href="https://commits.webkit.org/294061@main">https://commits.webkit.org/294061@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/37fbad6eeb521d30f6dd842aee515e81495b8ba2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/100726 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/20378 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/10677 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105863 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/51314 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/102767 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/20687 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/28852 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/76698 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/33737 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/103733 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15890 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90982 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57051 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/15699 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/8990 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/50690 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/85601 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9065 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108218 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27844 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20446 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/85657 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28207 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87184 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/85198 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21679 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29904 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/7632 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/21838 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/27779 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/33034 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/27590 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30908 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29148 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

beidson commented on Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
```diff
@@ -666,7 +666,10 @@ void NetworkResourceLoader::transferToNewWebProcess(NetworkConnectionToWebProces
     m_parameters.webFrameID = parameters.webFrameID;
     m_parameters.options.clientIdentifier = parameters.options.clientIdentifier;
-    m_parameters.options.resultingClientIdentifier = parameters.options.resultingClientIdentifier;
+//    m_parameters.options.resultingClientIdentifier = parameters.options.resultingClientIdentifier;
```
Remove.

youennf replied:
done

beidson reviewed (approved):
Remove the commented line, r+

youennf committed:
Client id mismatch after Cross-Origin-Opener-Policy Header
rdar://problem/149370100
https://bugs.webkit.org/show_bug.cgi?id=291315

Reviewed by Brady Eidson.

Before the patch, in case of process swapping, we would create a new DocumentLoader in the final process.
This would mean a new ScriptExecutionContextIdentifier and so a new clientId that is not is the same as the inital DocumentLoader
that triggered the request sent to the service worker.

To fix this, when process swap happens in NetworkResourceLoader::transferToNewWebProcess, we tell the new DocumentLoader that it
should reuse the initial DocumentLoader ScriptExecutionContextIdentifier (with the right process identifier).

We update NetworkResourceLoader, WebResourceLoader and DocumentLoader accordingly.
We then have to deal in SWServer with ScriptExecutionContextIdentifier that have the same UUID but not the same process identifier.

* LayoutTests/http/tests/workers/service/resultingClientId-worker.js: Added.
* LayoutTests/http/tests/workers/service/resultingClientId.https-expected.txt: Added.
* LayoutTests/http/tests/workers/service/resultingClientId.https.html: Added.
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::setNewResultingClientId):
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/workers/service/server/SWServer.cpp:
(WebCore::SWServer::registerServiceWorkerClient):
(WebCore::SWServer::unregisterServiceWorkerClient):
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::transferToNewWebProcess):
* Source/WebKit/WebProcess/Network/WebResourceLoader.cpp:
(WebKit::WebResourceLoader::updateResultingClientIdentifier):
* Source/WebKit/WebProcess/Network/WebResourceLoader.h:
* Source/WebKit/WebProcess/Network/WebResourceLoader.messages.in:

Canonical link: https://commits.webkit.org/294061@main
